File: 3208_alternating_groups_ii.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& w, int p) {
        int r = 0, t = 1, n = w.size();
        for (int i = 1; i < n + p - 1; i++) {
            t = (w[i % n] == w[(i - 1) % n]) ? 1 : t + 1;
            r += (t >= p);
        }
        return r;
    }
};

File: 3208_alternating_groups_ii.py
class Solution(object):
    def numberOfAlternatingGroups(self, z, p):
        """
        :type z: List[int]
        :type p: int
        :rtype: int
        """
        r, t, n = 0, 1, len(z)
        for i in range(1, n + p - 1):
            t = 1 if z[i % n] == z[(i - 1) % n] else t + 1
            r += t >= p
        return r
